.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,233 @@
+    required: false
+    default: temurin
+  java-version:
+    description: JDK version to install
+    required: false
+    default: "21"
+  maven-cache-key-modifier:
+    description: A modifier to use for maven cache key
+    required: false
+    default: "shared"
+  maven-mirrors:
+    description: |
+      JSON list of extra Maven mirrors.
+      It will be merged with Camunda Nexus settings (if enabled ), with precedence."
+    required: false
+    default: '[]'
+  maven-servers:
+    description: |
+      JSON list of extra Maven servers.
+      It will be merged with Camunda Nexus settings (if enabled), with precedence."
+    required: false
+    default: '[]'
+  maven-version:
+    description: |
+      Maven version to be used by the wrapper.
+      If not set, default version in .mvn/wrapper/maven-wrapper.properties will be used.
+    required: false
+  time-zone:
+    description: |
+      Time zone to set for the build environment. Use TZ identifiers, e.g., 'Europe/Berlin'.
+      https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+    required: false
+  vault-address:
+    description: Vault URL to retrieve secrets from
+    required: false
+  vault-role-id:
+    description: Vault Role ID to use
+    required: false
+  vault-secret-id:
+    description: Vault Secret ID to use
+    required: false
+
+outputs: {}
+
+runs:
+  using: composite
+  steps:
+  - name: Preliminary checks
+    id: checks
+    env:
+      DOCKERHUB: ${{ inputs.dockerhub }}
+      DOCKERHUB_READONLY: ${{ inputs.dockerhub-readonly }}
+      # Paths to Vault secrets for DockerHub accounts
+      DOCKERHUB_VAULT_PATHS: >-
+        {
+          "camunda": {
+            "path": "secret/data/products/camunda/ci/github-actions"
+          },
+          "operate": {
+            "path": "secret/data/products/operate/ci/github-actions"
+          },
+          "optimize": {
+            "path": "secret/data/products/optimize/ci/camunda-optimize"
+          },
+          "tasklist": {
+            "path": "secret/data/products/tasklist/ci/tasklist"
+          },
+          "zeebe": {
+            "path": "secret/data/products/zeebe/ci/zeebe"
+          }
+        }
+      REPO_OWNED_BY_CAMUNDA: ${{ github.repository_owner == 'camunda' }}
+      WORKFLOW_REF: ${{ github.workflow_ref }}
+    run: |
+      ## Preliminary checks
+
+      # Check if repo is owned by Camunda or an external fork
+      if [[ "${REPO_OWNED_BY_CAMUNDA}" != "true" ]]; then
+        echo "This repository does not belong to Camunda, and is probably an external fork."
+        echo "As a result, Vault secrets cannot be recovered, Camunda Nexus and DockerHub login will be disabled."
+      fi
+
+      # Check if DockerHub and DockerHub-readonly are both enabled
+      if [[ "${DOCKERHUB}" == "true" && "${DOCKERHUB_READONLY}" == "true" ]]; then
+        echo "Both dockerhub and dockerhub-readonly inputs are enabled. Only one can be enabled at a time."
+        exit 1
+      fi
```

Comment:
❓ Both variables could be combined into one 3-modal variable (`Dockerhub: yes/readonly/no`) to simplify/avoid this check without other downsides afaik.

Reply:
Good question.
On subsequent PR the `harbor` input of boolean type is introduced.  I am wondering it is not a bit confusing to have  two different type of variable ? e.g.:

```yaml
dockerhub: readonly
harbor: true
```

What do you htink ?

Reply:
I'm also fine with the current solution